geonode/people/adapters.py

```diff
@@ -347,23 +370,12 @@ def complete_login(self, request, app, token, response, **kwargs):
 
     def save_user(self, request, sociallogin, form=None):
         user = super(SocialAccountAdapter, self).save_user(request, sociallogin, form=form)
-        extractor = get_data_extractor(sociallogin.account.provider)
-        group_role_mapper = get_group_role_mapper(sociallogin.account.provider)
-        try:
-            groups = extractor.extract_groups(sociallogin.account.extra_data) or extractor.extract_roles(
-                sociallogin.account.extra_data
-            )
-
-            # check here if user is member already of other groups and remove it form the ones that are not declared here...
-            for groupprofile in user.group_list_all():
-                groupprofile.leave(user)
-            for group_role_name in groups:
-                group_name, role_name = group_role_mapper.parse_group_and_role(group_role_name)
-                groupprofile = GroupProfile.objects.filter(slug=group_name).first()
-                if groupprofile:
-                    groupprofile.join(user)
-                    if group_role_mapper.is_manager(role_name):
-                        groupprofile.promote(user)
-        except (AttributeError, NotImplementedError):
-            pass  # extractor doesn't define a method for extracting field
+        user = _update_user_groups_from_social(sociallogin, user)
         return user
+
+    def pre_social_login(self, request, sociallogin):
+        """update user groups on each social login"""
```

Comment:
Hi @sijandh35 why in the `pre_social_login` and not in the `populate_user`? Is because the `populate_user` is not called also during the login?